guides/non-tee.md

```diff
@@ -0,0 +1,245 @@
+
+- `$KBS_DIR_PATH/data/attestation-service/opa/policy.rego`: Policy file for evidence verification of AS, refer to [AS Policy Engine](https://github.com/confidential-containers/attestation-service#policy-engine) for more infomation.
+
+### Encrypting an Image
+
+[skopeo](https://github.com/containers/skopeo) is required to encrypt the container image.
+Follow the [instructions](https://github.com/containers/skopeo/blob/main/install.md) to install `skopeo`.
+
+Use `skopeo` to encrypt an image on the same node of the KBS cluster (use busybox:latest for example):
+
+```shell
+# edit ocicrypt.conf
+tee > ocicrypt.conf <<EOF
+{
+    "key-providers": {
+        "attestation-agent": {
+            "grpc": "127.0.0.1:50000"
+        }
+    }
+}
+EOF
+
+# encrypt the image
+OCICRYPT_KEYPROVIDER_CONFIG=ocicrypt.conf skopeo copy --insecure-policy --encryption-key provider:attestation-agent docker://library/busybox oci:busybox:encrypted
+```
+
+The image will be encrypted, and things happens in the KBS cluster background include:
+
+- CoCo Keyprovider generates a random key and a key-id. Then encrypts the image using the key.
+- CoCo Keyprovider registers the key with key-id into KBS.
+
+Then push the image to a registry:
+
+```shell
+skopeo copy oci:busybox:encrypted [SCHEME]://[REGISTRY_URL]:encrypted
+```
+Be sure to replace `[SCHEME]` with registry scheme type like `docker`, replace `[REGISTRY_URL]` with the desired registry URL like `docker.io/encrypt_test/busybox`.
+
+### Signing an Image
+
+[cosign](https://github.com/sigstore/cosign) is required to sign the container image. Follow the instructions here to install `cosign`:
+
+[cosign installation](https://docs.sigstore.dev/cosign/installation/)
+
+Generate a cosign key pair and register the public key to KBS storage:
+
+```shell
+cosign generate-key-pair
+mkdir -p $KBS_DIR_PATH/data/kbs-storage/default/cosign-key && cp cosign.pub $KBS_DIR_PATH/data/kbs-storage/default/cosign-key/1
+```
+
+Sign the encrypted image with cosign private key:
+
+```shell
+cosign sign --key cosign.key [REGISTRY_URL]:encrypted
+```
+
+Be sure to replace `[REGISTRY_URL]` with the desired registry URL of the encrypted image generated in previous steps.
+
+Then edit an image pulling validation policy file.
+Here is a sample policy file `security-policy.json`:
+
+```json
+{
+    "default": [{"type": "reject"}],
+    "transports": {
+        "docker": {
+            "[REGISTRY_URL]": [
+                {
+                    "type": "sigstoreSigned",
+                    "keyPath": "kbs:///default/cosign-key/1"
+                }
+            ]
+        }
+    }
+}
+```
+
+Be sure to replace `[REGISTRY_URL]` with the desired registry URL of the encrypted image.
+
+Register the image pulling validation policy file to KBS storage:
+
+```shell
+mkdir -p $KBS_DIR_PATH/data/kbs-storage/default/security-policy
+cp security-policy.json $KBS_DIR_PATH/data/kbs-storage/default/security-policy/test
```

Comment:
This looked a bit off, but I've never experimented with a validation policy. Is the `cp` command correct? It seems to copy but also rename `security-policy.json` to `test`. (Also, I couldn't find where this actually gets used, so I'm curious for a pointer if you have it.)

Reply:
I'm not sure about the command either. That content was just moved over from whatever was already in quickstart, so nothing in the non-tee or coco-dev guide was written by me.  If the command is wrong, we can fix it in a future pr.